### PR TITLE
407: Extended EmailDebrief with action_id

### PR DESCRIPTION
**Pull request by asgharsir**

### CODE REVIEW

File: participant/models.py [hunk 1]
```diff
@@ -85,19 +89,28 @@ def phishing_email_behaviour_and_actions(self, email_id):
         """
 
         def participant_behaviour(actions):
-            pb = ParticipantBehaviour.NEUTRAL
-            if [a for a in actions if a.get('action_type') in POSITIVE_ACTIONS]:
-                pb = ParticipantBehaviour.POSITIVE
-            if [a for a in actions if a.get('action_type') in NEGATIVE_ACTIONS]:
-                pb = ParticipantBehaviour.NEGATIVE
+            behaviour = ParticipantBehaviour.NEUTRAL
```
Comment (fericsepi) [inline participant/models.py]: please update the docstring to reflect what the function returns

File: participant/models.py [hunk 2]
```diff
@@ -85,19 +89,28 @@ def phishing_email_behaviour_and_actions(self, email_id):
...
+            action_id = None
+
+            for a in actions:
```
Comment (fericsepi) [inline participant/models.py]: let's use items() here instead
```python
for action_id, action_details in actions.items():
    ... 
```

File: participant/models.py [hunk 3]
```diff
@@ -85,19 +89,28 @@ def phishing_email_behaviour_and_actions(self, email_id):
...
             return pb
 
         actions = []
 
         if email_id in self.exercise.phishing_email_ids:
             for action_id, action_details in self.actions.items():
-                if email_id == action_details.get('email_id'):
-                    actions.append(action_details)
+                if email_id == action_details.get("email_id"):
+                    actions.append([action_id, action_details])
```
Comment (fericsepi) [inline participant/models.py]: Please include action_id in details instead as below:
```suggestion
                    action_details["action_id"] = action_id
                    actions.append(action_details)
```

File: participant/tests/test_api.py [hunk 4]
```diff
@@ -292,5 +292,14 @@ def test_participant_scores_debrief_behaviour_responses(self):
                 )
                 self.assertEqual(
                     behaviour,
-                    response.data["phishing_emails"][0]["participant_behaviour"],
+                    response.data["phishing_emails"][0]["participant_behaviour"][
+                        "behaviour"
+                    ],
                 )
+                if behaviour == "neutral":
+                    self.assertEqual(
```
Comment (fericsepi) [inline participant/tests/test_api.py]: - let's use `assertIsNone` instead for `neutral`
- for anything else let's include a check to ensure `action_id` in the response matches with our `action.id` (L281)

File: participant/models.py (hunk 2 above)
Comment (asgharsir) [reply participant/models.py]: It is condition based. So did the otherwise.

File: participant/models.py (hunk 3 above)
Comment (asgharsir) [reply participant/models.py]: Good. Avoided it because it was adding extra "action_id" item in all `participantActions` objects (in endpoint response).

**Comment by fericsepi** [comment]
#407

**Pull request merged**